src/openms/source/ANALYSIS/RNPXL/ModifiedPeptideGenerator.cpp

```diff
@@ -281,31 +338,33 @@ namespace OpenMS
     map<int, vector<const ResidueModification*> >::const_iterator pos_mod_it = map_compatibility.find(current_index);
     const vector<const ResidueModification*>& mods = pos_mod_it->second; // we don't need to check for .end as entry is guaranteed to exist
 
-    for (auto m : mods)
+    for (const ResidueModification* m : mods)
     {
+
       // copy peptide and apply modification
       AASequence new_peptide = current_peptide;
       if (current_index == C_TERM_MODIFICATION_INDEX)
       {
-        new_peptide.setCTerminalModification(m->getFullName());
+        new_peptide.setCTerminalModification(m);
       }
       else if (current_index == N_TERM_MODIFICATION_INDEX)
       {
-        new_peptide.setNTerminalModification(m->getFullName());
+        new_peptide.setNTerminalModification(m);
       }
       else
       {
-        new_peptide.setModification(current_index, m->getFullName());
+        const Residue* r = var_mods.at(m); // map modification to the modified residue
```

Comment:
any reason you use `at` and not direct access? Is it possible that the Residue is not found?

Reply:
Because .at is const access . Bracket is non-const